### PR TITLE
Add constraints around hawkular and heapster

### PR DESCRIPTION
**Pull request by wshearn**

### CODE REVIEW

**Review by twiest — APPROVED** [review]
These look fine to me, but I honestly know very little about the metrics deploy roles.

@kwoodson would you mind taking a look too?

File: ansible/roles/openshift_metrics/tasks/main.yml [hunk 1]
```diff
@@ -103,36 +103,92 @@
...
+    kind: rc
+    namespace: openshift-infra
+    replicas: 0
+
+- name: Wait for hawkular to scale down
+  pause:
+    seconds: 30
+  run_once: true
+
+- name: Add the resource constraints to the hawkular RC
```
Comment (kwoodson) [inline ansible/roles/openshift_metrics/tasks/main.yml]: Did you mean heapster?

File: ansible/roles/openshift_metrics/tasks/main.yml [hunk 2]
```diff
@@ -103,36 +103,92 @@
...
+  oc_edit:
+    kind: rc
+    name: "heapster"
+    namespace: openshift-infra
+    content:
+      spec.template.spec.containers[0].resources.limits.memory: "{{ osamet_heapster_limits_memory }}"
+      spec.template.spec.containers[0].resources.requests.memory: "{{ osamet_heapster_requests_memory }}"
+
+- name: scale up hawkular to 1
```
Comment (kwoodson) [inline ansible/roles/openshift_metrics/tasks/main.yml]: Did you mean heapster?

File: ansible/roles/openshift_metrics/tasks/main.yml [hunk 3]
```diff
@@ -103,36 +103,92 @@
...
+  oc_scale:
+    name: "heapster"
```
Comment (kwoodson) [inline ansible/roles/openshift_metrics/tasks/main.yml]: Don't need quotes on name.

File: ansible/roles/openshift_metrics/tasks/main.yml [hunk 4]
```diff
@@ -103,36 +103,92 @@
...
+
+- name: Wait for hawkular to scale down
+  pause:
+    seconds: 30
+  run_once: true
+
+- name: Add the resource constraints to the hawkular RC
+  oc_edit:
+    kind: rc
+    name: "heapster"
```
Comment (kwoodson) [inline ansible/roles/openshift_metrics/tasks/main.yml]: don't need quotes on name.

File: ansible/roles/openshift_metrics/tasks/main.yml [hunk 5]
```diff
@@ -103,36 +103,92 @@
...
+    replicas: 1
+
+- name: scale down heapster to 0
+  oc_scale:
+    name: "heapster"
+    kind: rc
+    namespace: openshift-infra
+    replicas: 0
+
+- name: Wait for hawkular to scale down
```
Comment (kwoodson) [inline ansible/roles/openshift_metrics/tasks/main.yml:172]: did you mean heapster?

File: ansible/roles/openshift_metrics/tasks/main.yml [hunk 6]
```diff
@@ -103,36 +103,92 @@
...
+    kind: rc
+    name: "hawkular-metrics"
+    namespace: openshift-infra
+    content:
+      spec.template.spec.containers[0].resources.limits.memory: "{{ osamet_hawkular_limits_memory }}"
+      spec.template.spec.containers[0].resources.requests.memory: "{{ osamet_hawkular_requests_memory }}"
+
+- name: scale up hawkular to 1
+  oc_scale:
+    name: "hawkular-metrics"
```
Comment (kwoodson) [inline ansible/roles/openshift_metrics/tasks/main.yml]: no need for quotes

File: ansible/roles/openshift_metrics/tasks/main.yml [hunk 7]
```diff
@@ -103,36 +103,92 @@
...
+    kind: rc
+    namespace: openshift-infra
+    replicas: 1
+
+- name: scale down heapster to 0
+  oc_scale:
+    name: "heapster"
```
Comment (kwoodson) [inline ansible/roles/openshift_metrics/tasks/main.yml]: no need for quotes.

**Review by kwoodson — COMMENTED** [review]
Few suggestions.

**Pull request closed without merging**